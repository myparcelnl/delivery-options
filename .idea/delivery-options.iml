<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager">
    <content url="file://$MODULE_DIR$">
      <excludeFolder url="file://$MODULE_DIR$/dist" />
      <excludeFolder url="file://$MODULE_DIR$/.yarn" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
      <excludeFolder url="file://$MODULE_DIR$/artifact" />
      <excludeFolder url="file://$MODULE_DIR$/.husky" />
      <excludeFolder url="file://$MODULE_DIR$/.nx" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>