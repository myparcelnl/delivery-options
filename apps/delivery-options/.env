VITE_CLASS_BASE=myparcel-delivery-options
