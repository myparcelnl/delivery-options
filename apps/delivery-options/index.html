<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport" />
    <title>MyParcel Delivery Options</title>
  </head>
  <body>
    <script type="importmap">
      {
        "imports": {
          "vue": "https://unpkg.com/vue@3/dist/vue.esm-browser.js"
        }
      }
    </script>

    <div id="myparcel-delivery-options"></div>

    <script src="/src/main.ts" type="module"></script>
  </body>
</html>
