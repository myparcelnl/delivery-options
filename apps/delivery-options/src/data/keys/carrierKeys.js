/** @type {MyParcel.CarrierName} */
export const POSTNL = 'postnl';

/** @type {MyParcel.CarrierName} */
export const DPD = 'dpd';

/** @type {MyParcel.CarrierName} */
export const BPOST = 'bpost';

/** @type {MyParcel.CarrierName} */
export const INSTABOX = 'instabox';

/** @type {MyParcel.CarrierName} */
export const CHEAP_CARGO = 'cheapcargo';

/** @type {MyParcel.CarrierName} */
export const DHL = 'dhl';

/** @type {MyParcel.CarrierName} */
export const DHL_FOR_YOU = 'dhlforyou';

/** @type {MyParcel.CarrierName} */
export const DHL_PARCEL_CONNECT = 'dhlparcelconnect';

/** @type {MyParcel.CarrierName} */
export const DHL_EUROPLUS = 'dhleuroplus';
