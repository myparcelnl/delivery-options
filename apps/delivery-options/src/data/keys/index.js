export * from './configKeys';
export * from './addressKeys';
export * from './carrierKeys';
export * from './stringsKeys';
export * from './countryCodes';
export * from './platformKeys';
export * from './settingsConsts';
export * from './packageTypeConfig';

export * as ADDRESS from './addressKeys';

export * as CARRIERS from './carrierKeys';

export * as CONFIG from './configKeys';

export * as CONSTS from './settingsConsts';

export * as STRINGS from './stringsKeys';

export * as PLATFORMS from './platformKeys';

