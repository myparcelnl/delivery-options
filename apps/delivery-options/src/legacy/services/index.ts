export * from './createScript';
export * from './directives';
