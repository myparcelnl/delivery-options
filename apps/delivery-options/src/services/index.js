export * from './checkSelector';
export * from './createScript';
export * from './cssClassBase';
export * from './directives';
export * from './filters';
export * from './polyfills';
export * from './showDeveloperInfo';
