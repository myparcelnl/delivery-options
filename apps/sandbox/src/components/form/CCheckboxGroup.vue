<template>
  <CInput
    v-model="selected"
    v-bind="filteredProps"
    component="BFormCheckboxGroup"
    :stacked="options.length >= 3"
    :options="translatedOptions" />
</template>

<script>
import { BFormCheckboxGroup } from 'bootstrap-vue/src/components/form-checkbox/form-checkbox-group';
import CInput from '../../delivery-options/src/sandbox/components/form/CInput';
import { formCheckboxGroup } from '../../delivery-options/src/sandbox/services/mixins/formCheckboxGroup';

export default {
  name: 'CCheckboxGroup',
  components: { CInput },
  extends: BFormCheckboxGroup,
  mixins: [
    formCheckboxGroup,
  ],
};
</script>
