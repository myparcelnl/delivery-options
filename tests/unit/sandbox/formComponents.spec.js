import { mockSandbox, shallowMockSandbox } from './mockSandbox';
import CCodeEditor from '@/sandbox/components/form/CCodeEditor';
import CCountrySelect from '@/sandbox/components/form/CCountrySelect';
import CCurrency from '@/sandbox/components/form/CCurrency';
import { i18n } from '@/sandbox/services/vue-i18n';
import CCheckboxGroup from '@/sandbox/components/form/CCheckboxGroup';
import { mockConfigBus } from '@Tests/unit/delivery-options/mockConfigBus';
import { DEFAULT_PLATFORM } from '@/data/keys/settingsConsts';
import { mockDeliveryOptions } from '@Tests/unit/delivery-options/mockDeliveryOptions';
import * as CONFIG from '@/data/keys/configKeys';
import { MYPARCEL, SENDMYPARCEL } from '@/data/keys/platformKeys';
import * as CARRIERS from '@/data/keys/carrierKeys';
import { DROP_OFF_DAYS } from '@/data/keys/configKeys';

// Test options.
const options = [
  { text: 'Option A', value: 'a' },
  { text: 'Option B', value: 'b' },
];

describe('Sandbox form components', () => {
  test('CCodeEditor', async() => {
    expect.assertions(9);

    /** @type {Wrapper<CCodeEditor>} */
    const wrapper = mockSandbox(null, null, CCodeEditor);
    const textarea = wrapper.find('textarea');

    const value = JSON.stringify({ data: { code: ['test code'] } });

    textarea.setValue(value);
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted().input[0]).toEqual([value]);

    await wrapper.vm.$nextTick();
    expect(wrapper.find('textarea').element).toHaveValue(
      '{"data":{"code":["test code"]}}',
    );

    const { element: codeElement } = wrapper.findByTestId('code');
    const { element: textElement } = wrapper.findByTestId('text');

    wrapper.vm.showCode();
    await wrapper.vm.$nextTick();
    expect(codeElement).toBeVisible();
    expect(textElement).not.toBeVisible();

    wrapper.vm.hideCode();
    await wrapper.vm.$nextTick();
    expect(codeElement).not.toBeVisible();
    expect(textElement).toBeVisible();

    wrapper.vm.showCode();
    await wrapper.vm.$nextTick();
    expect(codeElement).toBeVisible();
    expect(textElement).not.toBeVisible();

    textarea.setValue(['invalid data']);
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.valid).toBe(false);
  });

  test('CCountrySelect', async() => {
    expect.assertions(1);
    const wrapper = shallowMockSandbox(null, null, CCountrySelect);

    wrapper.vm.$i18n.locale = 'nl';
    wrapper.vm.$i18n.locale = 'en';
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.mutableOptions).toContainObject({
      text: 'Sweden',
      value: 'SE',
    });
  });

  test('CCurrency', () => {
    const wrapper = mockSandbox(
      null,
      {
        i18n,
        propsData: {
          value: '2.36',
        },
      },
      CCurrency,
    );

    expect(wrapper.find('input').element).toHaveValue(2.36);
  });

  test.each([
    ['@/sandbox/components/form/CCheckboxGroup', { options }],
    ['@/sandbox/components/form/CNumber'],
    ['@/sandbox/components/form/CRadioGroup', { options }],
    ['@/sandbox/components/form/CSelect', { options }],
    ['@/sandbox/components/form/CTextInput'],
    ['@/sandbox/components/form/CTextarea'],
    ['@/sandbox/components/form/CTimepicker'],
    ['@/sandbox/components/form/CToggle'],
  ])('%s', async(componentPath, propsData = {}) => {
    expect.assertions(1);
    const component = (await import(componentPath)).default;
    const wrapper = mockSandbox(null, { propsData }, component);

    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
