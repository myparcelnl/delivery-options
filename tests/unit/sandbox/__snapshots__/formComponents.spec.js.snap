// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sandbox form components @/sandbox/components/form/CCheckboxGroup 1`] = `
<bformcheckboxgroup-stub
  checked=""
  disabledfield="disabled"
  htmlfield="html"
  options="[object Object],[object Object]"
  textfield="text"
  value=""
  valuefield="value"
/>
`;

exports[`Sandbox form components @/sandbox/components/form/CNumber 1`] = `
<bforminput-stub
  debounce="0"
  formatter="function bvConfigurablePropDefault() {
    const value = (0, _config.getComponentConfig)(componentKey, key, prop.default);
    return (0, _inspect.isFunction)(value) ? value() : value;
  }"
  lazy="true"
  placeholder=<Default>
  step="1"
/>
`;

exports[`Sandbox form components @/sandbox/components/form/CRadioGroup 1`] = `
<div
  class="bv-no-focus-ring"
  options="[object Object],[object Object]"
  role="radiogroup"
  tabindex="-1"
>
  <div
    class="custom-control custom-control-inline custom-radio"
  >
    <input
      class="custom-control-input"
      type="radio"
      value="a"
    />
    <label
      class="custom-control-label"
    >
      <span>
        Option A
      </span>
    </label>
  </div>
  <div
    class="custom-control custom-control-inline custom-radio"
  >
    <input
      class="custom-control-input"
      type="radio"
      value="b"
    />
    <label
      class="custom-control-label"
    >
      <span>
        Option B
      </span>
    </label>
  </div>
</div>
`;

exports[`Sandbox form components @/sandbox/components/form/CSelect 1`] = `
<bformselect-stub
  options="[object Object],[object Object]"
/>
`;

exports[`Sandbox form components @/sandbox/components/form/CTextInput 1`] = `
<bforminput-stub
  lazy="true"
  placeholder=<Default>
/>
`;

exports[`Sandbox form components @/sandbox/components/form/CTextarea 1`] = `
<textarea
  class="form-control"
  debounce="0"
  formatter="function bvConfigurablePropDefault() {
      var value = getComponentConfig(componentKey, key, prop.default);
      return isFunction(value) ? value() : value;
    }"
  rows="2"
  wrap="soft"
/>
`;

exports[`Sandbox form components @/sandbox/components/form/CTimepicker 1`] = `
<binput-stub
  lazy="true"
  placeholder="HH:mm"
/>
`;

exports[`Sandbox form components @/sandbox/components/form/CToggle 1`] = `
<bformcheckbox-stub
  id=""
  size="lg"
  switch=""
  type="text"
/>
`;
